File: scripts/extract_players.js
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';
import { dirname } from 'path';

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

const sourcePath = path.join(__dirname, '..', 'dist', 'assets', 'index-BH2pbF1J.js');
const text = fs.readFileSync(sourcePath, 'utf8');

const start = text.indexOf('be={');
if (start === -1) {
  throw new Error("Could not find 'be={' in built JS file");
}

let stack = [];
let end = -1;
for (let i = start; i < text.length; i++) {
  const ch = text[i];
  if (ch === '{') {
    stack.push(i);
  } else if (ch === '}') {
    stack.pop();
    if (stack.length === 0) {
      end = i;
      break;
    }
  }
}

if (end === -1) {
  throw new Error('Could not find end of object starting at be={');
}

const obj = text.slice(start, end + 1);
const outPath = path.join(__dirname, '..', 'data', 'extractedPlayersBase.js');
fs.writeFileSync(outPath, 'export const EXTRACTED = ' + obj + ';\n', 'utf8');
console.log(`Wrote ${outPath} (length ${obj.length}).`);

File: scripts/extract_players.py
import pathlib

# Path to the built JS file containing the data.
source_path = pathlib.Path(r"c:\Users\User\Desktop\rb-scout\dist\assets\index-BH2pbF1J.js")
text = source_path.read_text(encoding="utf-8", errors="replace")

start = text.find("be={")
if start == -1:
    raise SystemExit("Could not find 'be={' in built JS file")

# Find the matching closing brace for the object.
stack = []
end = None
for i, ch in enumerate(text[start:], start):
    if ch == "{":
        stack.append(i)
    elif ch == "}":
        stack.pop()
        if not stack:
            end = i
            break

if end is None:
    raise SystemExit("Could not find end of object starting at 'be={'")

obj = text[start:end+1]

out_path = pathlib.Path(r"c:\Users\User\Desktop\rb-scout\data\extractedPlayersBase.js")
out_path.write_text("export const EXTRACTED = " + obj + ";\n", encoding="utf-8")
print(f"Wrote {out_path} (length {len(obj)}).")
